README.md

```diff
@@ -1,14 +1,40 @@
-# dcos-go - DC/OS golang shared libraries
+# dcos-go - DC/OS golang shared packages
 [![Build Status](https://jenkins.mesosphere.com/service/jenkins/buildStatus/icon?job=public-dcos-go/public-dcos-go-master)](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-go/job/public-dcos-go-master/)
 [![GoDoc](https://godoc.org/github.com/dcos/dcos-go?status.svg)](https://godoc.org/github.com/dcos/dcos-go)
 [![Go Report Card](https://goreportcard.com/badge/github.com/dcos/dcos-go)](https://goreportcard.com/report/github.com/dcos/dcos-go)
 
+This repository exists to collect commonly shared Go code within Mesosphere. It can include code that is specific to DC/OS and can also include code that is more utilitarian in nature.
+
+## Contributing
+
+Everyone is welcome to submit Pull Requests.  
+
+If you find yourself writing the same code more than once in separate projects, or have seen the same code in more than one project, it may be a good candidate for inclusion into this repository.  Be sure to ask on the [mailing list](https://groups.google.com/a/mesosphere.io/forum/#!forum/go) if you are unsure before contributing.
+
+Pull requests should adhere to the following guidelines:
+
+* Reviewer first: create your changeset with care for the reviewer. 
+* The changeset should have a good description. 
+* Each commit in the changeset should have a description as well as be focused on one thing.
+* Code must be tested
+* Code should be written with care to allow for non-breaking changes going forward
+* If introducing a new package, the package should have a descriptive README.
+* Try not to introduce external dependencies unless necessary.
+
+[Collin](mailto:cvandyck@mesosphere.com) is currently the temporary maintainer of this project. If you have questions about contributing please feel free to contact him or post on the mailing list.
+
+## External Libraries
+
+It's somewhat difficult to only rely on the Go stdlib when creating shared libraries.  Certainly if one is already used in the repository do not introduce a new one that does the same thing if at all possible.
+
+Going forward we will be creating guidelines for vendoring dependencies in this project.  
+
 ## Packages In This Library
```

Comment:
might it make sense for README.md to be generated from a README.md.template file, by some process that examines which golang packages are present in this repo (and then a generator is run to build a correct README.md)?